src/lazy/expanded/macro_evaluator.rs

```diff
@@ -134,35 +141,53 @@ impl<'top, D: Decoder> MacroExpr<'top, D> {
         self
     }
 
-    pub fn expand(&self, environment: Environment<'top, D>) -> IonResult<MacroExpansion<'top, D>> {
+    pub fn expand(&self) -> IonResult<MacroExpansion<'top, D>> {
         match self.kind {
-            MacroExprKind::TemplateMacro(t) => t.expand(environment),
+            MacroExprKind::TemplateMacro(t) => t.expand(),
+            MacroExprKind::TemplateArgGroup(g) => g.expand(),
             MacroExprKind::EExp(e) => e.expand(),
-            MacroExprKind::EExpArgGroup(g) => g.expand(environment),
+            MacroExprKind::EExpArgGroup(g) => g.expand(),
```

Comment:
🗺️ As mentioned prior, each of these types now captures a reference to the correct `Environment` and no longer needs one to be passed in.